File: clients/javascript/src/models/msgStreamCommitIn.ts
// this file is @generated

export interface MsgStreamCommitIn {
    topic: string;
    consumerGroup: string;
    offset: number;
}

export const MsgStreamCommitInSerializer = {
    _fromJsonObject(object: any): MsgStreamCommitIn {
        return {
            topic: object['topic'],
            consumerGroup: object['consumer_group'],
            offset: object['offset'],
            };
    },

    _toJsonObject(self: MsgStreamCommitIn): any {
        return {
            'topic': self.topic,
            'consumer_group': self.consumerGroup,
            'offset': self.offset,
            };
    }
}

File: clients/rust/src/models/msg_stream_commit_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct MsgStreamCommitIn {
    pub topic: String,

    pub consumer_group: String,

    pub offset: u64,
}

impl MsgStreamCommitIn {
    pub fn new(topic: String, consumer_group: String, offset: u64) -> Self {
        Self {
            topic,
            consumer_group,
            offset,
        }
    }
}
